app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,75 @@
+class RecipesController < ApplicationController
+  load_and_authorize_resource except: :public_recipes
+  before_action :set_recipe, only: %i[show edit update destroy]
+
+  # GET /recipes or /recipes.json
+  def index
+    @recipes = Recipe.all
+  end
+
+  def public_recipes
+    @recipes = Recipe.where({ public: true }).order(created_at: :desc)
+  end
+
```

Comment:
- [ ] We don't implement SQL Queries inside the controller, we use the model to keep the SQL queries implementation.